### DISCUSSION TITLE
0.3.3 Minor upgrade with new API

### DISCUSSION THREAD

**Discussion by berrystephenw** [berrystephenw/OctoText#119] [General]
Added features for various users:

- time based notifications (ie notifications every 10 minutes)
- username only authentication
- bugfix for webcam timeouts
- new API interface as requested (and coded) by @OllisGit 
- general code cleanup

Any issues or regressions please let me know and I'll do my best to get them fixed quickly. 

**Full Changelog**: https://github.com/berrystephenw/OctoText/compare/0.3.1...main

<hr /><em>This discussion was created from the release <a href='https://github.com/berrystephenw/OctoText/releases/tag/0.3.3'>0.3.3 Minor upgrade with new API</a>.</em>

**Comment by sliverbaer** [comment]
Howdy,

After the update my settings were wiped out.  After entering them back in it doesn't seem to be working.  Had been using smtp.office365.com for an outlook.com account.

"Test message problem!
Exception while talking to your mail server, check your SMTP settings."

**Comment by berrystephenw** [comment]
Hi - I'll need to see a copy of the logs. Can you attach them? Feel free to edit the file if you want to hide your email address (everything else is pretty generic). You will need to enable debug level logging for this, let me know if you need help with that.

**Reply by sliverbaer** [reply]
Hmmm, will it require a restart after I have enabled it under Settings > Logging > octoprint.plugins.OctoText ?

I don't see a log with that name in the webgui or under ~/.octoprint/logs

**Comment by berrystephenw** [comment]
Yes you will have to restart - I don't remember having to do that before, might be new behavior with the latest octoprint. There isn't an individual log for octotext, it just puts more messages in the octoprint.log file.

**Reply by sliverbaer** [reply]
Ok, I see the debugging in the log now.  It is attached.
[octoprint2.log](https://github.com/berrystephenw/OctoText/files/8239033/octoprint2.log)
.

**Reply by sliverbaer** [reply]
I actually figured it out after looking at the log.  The field in the settings shows username and then in the other box you enter the domain, but it wasn't appearing that way in the log.  I changed the first box to be username@domain and ignored the 2nd box and it worked fine.  I guess the option above is confusing if you have it checked.  (unchecking and then filling the username in first box and then domain in second box works fine.)
![2022-03-13 00_45_11-OctoPrint](https://user-images.githubusercontent.com/101478667/158046874-f2bbf451-2fa0-4ed0-8258-ab0c15070467.png)

Sorry, I guess that checkbox had me confused.

**Reply by berrystephenw** [reply]
Ok good you figured it out - yes the checkbox is only for certain situations where the mail server doesn't need or want the gateway address.
